src/statisticTab.ts

```diff
@@ -0,0 +1,91 @@
+
+/* eslint-disable @typescript-eslint/naming-convention */
+import * as vscode from "vscode";
+import { basename } from "path";
+import {
+    EVENT_NAMES_TO_STATISTIC,
+  } from "refact-chat-js/dist/events";
+import * as fetchAPI from "./fetchAPI";
+import { v4 as uuidv4 } from "uuid";
+
+export class StatisticTab {
+    public constructor(
+        public web_panel: vscode.WebviewPanel | vscode.WebviewView,
+    ) {
+        this.handleEvents = this.handleEvents.bind(this);
+        this.web_panel.webview.onDidReceiveMessage(this.handleEvents);
+    }
+
+    private handleEvents(message: any) {
+        switch (message.type) {
```

Comment:
Is there an error case?

Reply:
hm, seems I was checking smth and forgot to put that case  again, will do it

Reply:
oh wait it's in the catch... I wonder why i wasn't getting an error message when there was a 500 response